Doc/library/shutil.rst

```diff
@@ -680,6 +681,33 @@ The resulting archive contains:
     -rw-r--r-- tarek/staff   37192 2010-02-06 18:23:10 ./known_hosts
 
 
+.. _shutil-archiving-example-with-basedir:
+
+Archiving example with *base_dir*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this example, we have a file :file:`please_add.txt` within the directory
+structure :file:`tmp/root/structure/content`, which we want to add to the
+archive along with the directories :file:`structure` and :file:`content`, and a
+second file :file:`do_not_add.txt` in :file:`tmp/root/structure`, which we don't
+want to have in the archive. Therefore we use the following::
+
+    >>> from shutil import make_archive
+    >>> make_archive(base_name='tmp/archive',
```

Comment:
`base_name` and `format` are required. I'd just pass them as positional-only parameters (and to be honest, `base_name` is a bit misleading) Also, please use hangng indentation:

```py
make_archive(
    'my_archive',
    'tar',
    root_dir='tmp/root',
    base_dir='structure/content/',
)
```

Reply:
By the way, creating the archive file outside of `/tmp` would make the example easier to understand.